packages/uilib/src/lib/plugins/type-designer/components/controls/get-node-name.ts

```diff
@@ -0,0 +1,9 @@
+import { BayType, IEDType, LDeviceType, SubstationType, VoltageLevelType } from "../../types"
+
+export function getNodeName(node: BayType | IEDType | LDeviceType | VoltageLevelType | SubstationType): string {
+    if ('inst' in node) return 'LDeviceType';
+    if ('manufacturer' in node) return 'IEDType';
+    if ('nomFreq' in node) return 'VoltageLevelType';
+    if ('name' in node && 'desc' in node) return 'BayType';
+    return 'SubstationType';
+}
```

Comment:
This does not seem stable, the properties of the Types might change later on.

I suggest to add an Enum with all types (this could also be used when building the Xml Elements) and set a property to the enum value in the node.

Reply:
Logic on the core side.